### PR TITLE
Add mutation to change the authenticated user's password

### LINKED ISSUE #4287: Add mutation to allow authenticated user to change password
### What I'm trying to achieve
~~I would like to use setPassword mutation for changing users password but I need one time token and there is no query right now for obtaining it.~~

~~setPassword mutation receives:
    token String! - A one-time token required to set the password
    password String! - Password~~

There is need for new mutation that will allow change of password for authenticated user that will take only new password.

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #4287 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 2a3927c499961c5b07b73c48aff9803e2a82a775 (mirumee/saleor @ 4287/Logged-in_user_password_change)
Base comparison is de8091e0fcbf7c7f102637edeaea1ab34d58e016.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         48	         48	             24

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	         15	         15	              8
```

</p>
</details>

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Remove dashboard 2.0 files - #4631 by @dominik-zeglen
 - Refactoring of password recovery system - #4617 by @fowczarek
 - Added capability to filter attributes per global ID - #4640 by @NyanKiyoshi.
+- Add mutation for change authenticated user password - #4656 by @fowczarek
```
Comment (NyanKiyoshi) [inline CHANGELOG.md]:
```suggestion
- Add mutation to change the authenticated user's password - #4656 by @fowczarek
```

File: saleor/account/__init__.py [hunk 2]
```diff
@@ -8,6 +8,7 @@ class CustomerEvents:
     ACCOUNT_CREATED = "account_created"
     PASSWORD_RESET_LINK_SENT = "password_reset_link_sent"
     PASSWORD_RESET = "password_reset"
+    PASSWORD_CHANGE = "password_change"
```
Comment (NyanKiyoshi) [inline saleor/account/__init__.py]:
```suggestion
    PASSWORD_CHANGED = "password_changed"
```

File: saleor/account/events.py [hunk 3]
```diff
@@ -25,6 +25,10 @@ def customer_password_reset_event(*, user: UserType) -> Optional[CustomerEvent]:
     return CustomerEvent.objects.create(user=user, type=CustomerEvents.PASSWORD_RESET)
 
 
+def customer_password_change_event(*, user: UserType) -> Optional[CustomerEvent]:
```
Comment (NyanKiyoshi) [inline saleor/account/events.py]:
```suggestion
def customer_password_changed_event(*, user: UserType) -> Optional[CustomerEvent]:
```

File: saleor/graphql/account/mutations/base.py [hunk 4]
```diff
@@ -114,6 +114,41 @@ def perform_mutation(cls, _root, info, **data):
...
+        if not user.check_password(old_password):
+            raise ValidationError({"old_password": "Old password isn't valid."})
+        try:
+            password_validation.validate_password(new_password, user)
+        except ValidationError as error:
+            raise ValidationError({"new_password": error.messages})
+
+        user.set_password(new_password)
+        user.save(update_fields=["password"])
+        account_events.customer_password_change_event(user=user)
```
Comment (NyanKiyoshi) [inline saleor/graphql/account/mutations/base.py]:
```suggestion
        account_events.customer_password_changed_event(user=user)
```

File: saleor/graphql/account/mutations/base.py [hunk 5]
```diff
@@ -114,6 +114,41 @@ def perform_mutation(cls, _root, info, **data):
         return RequestPasswordReset()
 
 
+class PasswordChange(BaseMutation):
+    user = graphene.Field(User, description="An user instance with new password.")
```
Comment (maarcingebala) [inline saleor/graphql/account/mutations/base.py]:
```suggestion
    user = graphene.Field(User, description="A user instance with a new password.")
```

**Pull request merged**